lib/express.js

```diff
@@ -32,16 +57,22 @@ exports = module.exports = createApplication;
  * @api public
  */
 
-function createApplication() {
+function createApplication(opts) {
+  opts = opts || {}
+  opts.reqProto = opts.reqProto || http.IncomingMessage.prototype
+  opts.resProto = opts.resProto || http.ServerResponse.prototype
+  opts.createServer = opts.createServer || http.createServer
```

Comment:
Usually not best practice to alter the object someone is passing in; the object could potentially be frozen, even, causing an exception here. `var createServer = opts.createServer || http.createServer` is probably fine (and then changing the vars below).